### PR TITLE
[CARBONDATA-628] Fixed measure selection with out table order gives wrong result with vectorized reader enabled

### PR DESCRIPTION
**Pull request by ravipesala**
If the table is created with measure order like m1, m2 and user selects the measures m2, m1 then it returns wrong result with vectorized reader enabled

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/561/

File: core/src/main/java/org/apache/carbondata/scan/processor/AbstractDataBlockIterator.java [hunk 1]
```diff
@@ -85,12 +90,15 @@ public AbstractDataBlockIterator(BlockExecutionInfo blockExecutionInfo,
       blockletScanner = new NonFilterScanner(blockExecutionInfo, queryStatisticsModel);
     }
     if (blockExecutionInfo.isRawRecordDetailQuery()) {
+      LOGGER.audit("Row based raw collector is used to scan and collect the data");
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/scan/processor/AbstractDataBlockIterator.java]: Should it be audit or info? Audit is used for keeping operational log, right?

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/563/

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**